lib/xandra/retry_strategy.ex

```diff
@@ -16,37 +16,51 @@ defmodule Xandra.RetryStrategy do
   When a query fails and a retry strategy module was passed as an option, Xandra
   will:
 
-    1. invoke the `c:new/1` callback with the options passed to the failing
-       function to initialize the given retry strategy
+    1. invoke the `c:new/2` callback with the currently connected hosts and options
```

Comment:
If we put the currently-connected hosts in the options, we don't need to change the signature and break backwards compatibility.

Reply:
I think this is a breaking change since we need host information from the client now to know which host to execute on and it is okay if it is not backwards compatible anymore. 

I think passing this as an argument and make the client explicitly deal with connections and hosts to save to the state makes it clearer. It is clear that we only get the list of connected hosts on initialization and whatever information we're saving to the state can be outdated during the course of execution.